osu.Game/Graphics/Containers/LinkFlowContainer.cs

```diff
@@ -123,7 +126,7 @@ protected override IEnumerable<Drawable> CreateDrawablesFor(TextFlowContainer te
             }
         }
 
-        protected virtual DrawableLinkCompiler CreateLinkCompiler(ITextPart textPart) => new DrawableLinkCompiler(textPart);
+        protected virtual DrawableLinkCompiler CreateLinkCompiler(ITextPart textPart) => new DrawableLinkCompiler(textPart, currentLink);
```

Comment:
This field usage is super super dodgy.

Instead, I think you should add a `LinkDetails` field in `TextLink` and pass it in in the constructor at:

https://github.com/ppy/osu/blob/7bc8908ca9c026fed1d831eb6e58df7624a8d614/osu.Game/Graphics/Containers/LinkFlowContainer.cs#L92

You can then consume it in the place you are currently.

Reply:
Should be addressed in https://github.com/ppy/osu/pull/23334/commits/e92572b62792b92fef9803a85293000b82f5836a.